flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ShowDatabasesOperation.java

```diff
@@ -20,26 +20,108 @@
 
 import org.apache.flink.annotation.Internal;
 import org.apache.flink.table.api.internal.TableResultInternal;
+import org.apache.flink.table.functions.SqlLikeUtils;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
 import static org.apache.flink.table.api.internal.TableResultUtils.buildStringArrayResult;
 
 /** Operation to describe a SHOW DATABASES statement. */
 @Internal
 public class ShowDatabasesOperation implements ShowOperation {
 
+    private final String preposition;
```

Comment:
Is there a difference whether it is `FROM` or `IN` ? If there is no difference, could we parse both, but use one of the two when unparsing? We wouldn't need to store the original preposition.

Reply:
I see what you mean. IMO we need to keep `preposition` for 

- the correct summary string (`ShowDatabasesOperation::asSummaryString`)
- therefore, unparsing accordingly

Reply:
I don't get the argumentation. Could you please elaborate? `asSummaryString` prints a summary of the operation, I don't see why we need the exact same preposition that was used. I could even see the summary as `("ShowDatabases(catalog='abc', like='...', ....)`.

Even more I don't see a problem with unparsing to a different preposition as long as the effect is the same. In the end it makes no difference at all which one was used.

Reply:
If we are flexible with the summary string, then I completely agree